tempesta_fw/http_msg.c

```diff
@@ -498,15 +546,18 @@ __hdr_sub(TfwHttpMsg *hm, char *name, size_t n_len, char *val, size_t v_len,
 		.flags = 3 << TFW_STR_CN_SHIFT
 	};
 
+	if (unlikely(TFW_STR_DUP(orig_hdr)))
+		return __hdr_sub_dup(hm, orig_hdr, &hdr, hid);
```

Comment:
Again I agree with @keshonok: if the first header of duplicate string is long enough to place the new header, then we can rewrite the header at the first duplicate string regardless other duplicated of original header (whether they're at the same skb or not). The optimization should be small and straightforward. Probably, it has sense to handle the case in `__hdr_sub_dup()`.